src/System.Numerics.Tensors/System/Numerics/TensorView.cs

```diff
@@ -0,0 +1,146 @@
+
+                _backingTensor.SetValue(_offset + ArrayUtilities.GetIndex(_stridesIntoBackingTensor, indices), value);
+            }
+        }
+
+        public override T GetValue(int index)
+        {
+            EnsureValidIndex(index);
+
+            Span<int> indices = Rank < ArrayUtilities.StackallocMax ? stackalloc int[Rank] : new Span<int>(new int[Rank]);
+            ArrayUtilities.GetIndices(Strides, IsReversedStride, index, indices);
+
+            return this[indices];
+        }
+
+        public override void SetValue(int index, T value)
+        {
+            EnsureValidIndex(index);
+
+            Span<int> indices = Rank < ArrayUtilities.StackallocMax ? stackalloc int[Rank] : new Span<int>(new int[Rank]);
+            ArrayUtilities.GetIndices(Strides, IsReversedStride, index, indices);
+
+            this[indices] = value;
+        }
+
+        public override Tensor<T> Clone()
+        {
+            Tensor<T> result = _backingTensor.CloneEmpty(Dimensions);
+
+            // TODO: don't use the linearized index here since we always need to re-calculate the indices.
+            // Instead, we should have an enumerator that works on indices.
+            for (int i = 0; i < Length; i++)
+            {
+                result.SetValue(i, GetValue(i));
+            }
+
+            return result;
+        }
+
+        public override Tensor<TResult> CloneEmpty<TResult>(ReadOnlySpan<int> dimensions)
+        {
+            return _backingTensor.CloneEmpty<TResult>(dimensions);
+        }
+
+        public override Tensor<T> Reshape(ReadOnlySpan<int> dimensions)
+        {
+            if (dimensions.Length == 0)
+            {
+                throw new ArgumentException("Dimensions must contain elements.", nameof(dimensions));
+            }
+
+            var newSize = ArrayUtilities.GetProduct(dimensions);
+
+            if (newSize != Length)
+            {
+                throw new ArgumentException($"Cannot reshape array due to mismatch in lengths, currently {Length} would become {newSize}.", nameof(dimensions));
+            }
+
+            Tensor<T> result = _backingTensor.CloneEmpty(dimensions);
+
+            // TODO: don't use the linearized index here since we always need to re-calculate the indices.
+            // Instead, we should have an enumerator that works on indices.
+            for (int i = 0; i < Length; i++)
+            {
+                result.SetValue(i, GetValue(i));
+            }
+
+            return result;
+        }
+
+        private void EnsureValidIndices(ReadOnlySpan<int> indices)
+        {
+            if (indices.Length != Rank)
+            {
+                throw new ArgumentException($"{nameof(indices)} Length '{indices.Length}' must match Rank '{Rank}' of the current Tensor.", nameof(indices));
+            }
+
+            for (int i = 0; i < indices.Length; i++)
+            {
+                if (indices[i] < 0 || indices[i] >= Dimensions[i])
```

Comment:
Store indiced[i] in a local?

Reply:
Thats the least of my worries given the loop above uses signed integers...